plugins/DataLoadRlog/rlog_parser.cpp

```diff
@@ -69,13 +69,30 @@ bool RlogMessageParser::parseMessageImpl(const std::string& topic_name, capnp::D
     case capnp::DynamicValue::STRUCT: 
     {
       auto structValue = value.as<capnp::DynamicStruct>();
-      for (auto field : structValue.getSchema().getFields()) 
-      {
-        if (structValue.has(field))
-        {
+      if (is_root) {  // add global fields to each sub-struct (valid, logMonoTime, etc.)
+        std::string struct_name;
+        KJ_IF_MAYBE(e_, structValue.which()) { struct_name = e_->getProto().getName(); }
+        for (auto field : structValue.getSchema().getNonUnionFields()) {
           std::string name = field.getProto().getName();
-          if (show_deprecated || name.find("DEPRECATED") == std::string::npos) {
-            parseMessageImpl(topic_name + '/' + name, structValue.get(field), time_stamp, show_deprecated);
+          parseMessageImpl(topic_name + '/' + struct_name + "/event_" + name, structValue.get(field), time_stamp, false, show_deprecated);
+        }
+
+        for (auto field : structValue.getSchema().getUnionFields()) {
```

Comment:
Ideally we would iterate through all fields using `getFields` and then test if each field is a root field of Event, or the struct in the unnamed union. Then we could add root fields to all nested structs.